couchbase-lite/src/query.rs

```diff
@@ -58,16 +58,18 @@ impl Query<'_> {
             .ok_or_else(|| c4err.into())
     }
 
-    pub fn set_json_parameters<T>(&self, parameters: &T) -> Result<()> where T: Serialize {
+    pub fn set_json_parameters<T>(&self, parameters: &T) -> Result<()>
+    where
+        T: Serialize,
+    {
         let param_string = serde_json::to_string(parameters)?;
         let param_slice = param_string.as_bytes().as_flslice();
+        let param_copy = unsafe { FLSlice_Copy(param_slice) };
+        let param_copy: FlSliceOwner = param_copy.into();
```

Comment:
Is it working? FlSliceOwner type created for resource management, 
so it would call FLSlice_Release at exit of function, is it want we want?
I thought c4query_setParameters take care about freeing resources by it self,
if not why call `std::mem::forget(param_string)` in the first version of patch?

Reply:
Ah yes there's a Drop impl there...
I'll have to back off a little until I understand better.

The (misguided) reasoning for the first version was : I'll provide c4query_setParameters with a buffer I allocated rust-side but I don't want to deallocate it so I forget it. You pointed out the problems (both sides use different allocators).

So in the second version, I tried asking the library side to allocate a _copy_ of the query string, letting the library side free up the copy they allocated themselves and (auto) freeing the original. But I forgot about the Drop impl. I must _not_ make my slice copy into an owned one.

Reply:
I think I just probably need to implement ASFlSlice for FLSliceResult.